lib/streamlit/runtime/state/session_state.py

```diff
@@ -606,6 +608,14 @@ def get_stats(self) -> list[CacheStat]:
         stat = CacheStat("st_session_state", "", asizeof(self))
         return [stat]
 
+    def _check_serializable(self) -> None:
+        for k in self:
+            pickle.dumps(self[k])
```

Comment:
Let's catch pickle errors here and re-raise them in our own exception, like `UnserializableSessionStateError` or whatever. (And get Josh/product's input on nice error text.)